src/metrics/nodemetadata/metadata_req.go

```diff
@@ -0,0 +1,50 @@
+package nodemetadata
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/protolambda/rumor/p2p/rpc/methods"
+	"github.com/protolambda/rumor/p2p/rpc/reqresp"
+	"github.com/protolambda/zrnt/eth2/beacon"
+)
+
+// Function that opens a new Stream from the given host to send a RPC requesting the BeaconStatus of the given peer.ID
+// Returns the BeaconStatus of the given peer if succeed, error if failed
+func ReqBeaconStatus(ctx context.Context, h host.Host, peerID peer.ID) (data beacon.Status, err error) {
+	// Generate the compression
+	comp := reqresp.SnappyCompression{}
+	// Generate the Server Error Code
+	var resCode reqresp.ResponseCode // error by default
+	err = methods.StatusRPCv1.RunRequest(ctx, h.NewStream, peerID, comp,
+		reqresp.RequestSSZInput{Obj: &beacon.Status{}}, 1,
+		func() error {
+			return nil
+		},
+		func(chunk reqresp.ChunkedResponseHandler) error {
+			resCode = chunk.ResultCode()
+			switch resCode {
+			case reqresp.ServerErrCode, reqresp.InvalidReqCode:
+				msg, err := chunk.ReadErrMsg()
+				if err != nil {
+					return err
+				}
+				return errors.New(fmt.Sprintf("error requesting BeaconState RPC: %s", msg))
```

Comment:
can we wrap all errors so that we get the whole trace? `return errors.Wrap(err, "new error")`